app/controllers/api/assignments_controller.rb

```diff
@@ -5,50 +5,56 @@ class Api::AssignmentsController < ApplicationController
 
     def create
         authorize Assignment
-
-        created_assignments = []
-        created_action_items = []
-        participant_ids = bulk_assignment_params.fetch(:participant_ids, [])
-        action_items = bulk_assignment_params.fetch(:assignments, [])
-
-        if action_items.empty? || participant_ids.empty?
-            render json: { error: 'Action items and Participants must be populated'}, status: :unprocessable_entity
+        participant_ids_string = single_assignment_params.fetch(:participant_ids)
+        participant_ids = participant_ids_string.split(/,/)
+        puts participant_ids
+        if  participant_ids.empty?
+            render json: { error: 'Participant must be populated'}, status: :unprocessable_entity
             return
         end
-
-        action_items.each do |action_item|
-            begin
-                due_date = DateTime.parse(action_item[:due_date])
-            rescue ArgumentError
-                due_date = nil
-            end
-            action_item = ActionItem.new(action_item.except(:due_date)) 
-            template_sentry_helper(action_item)
-            action_item[:is_template] = false
-            if !participant_ids.empty? && action_item.save
-                created_action_items.append(action_item)
-                prepare_bulk_assignment(participant_ids, action_item, due_date).each do |assignment|
-                    assignment_sentry_helper(assignment)  
-                    if assignment.save
-                        AssignmentMailer.with(assignment: assignment, action_item: action_item).new_assignment.deliver_now
-                        created_assignments.append(assignment)
-                    else 
-                        action_item.destroy
-                        created_action_items.each {|item| item.destroy}
-                        Raven.capture_message("Could not create action item")
-                        render json: { error: 'Could not create action item' }, status: :unprocessable_entity
-                        return
-                    end
+        begin
+            due_date = DateTime.parse(single_assignment_params.fetch(:due_date))
+        rescue ArgumentError
+            due_date = nil
+        end 
+        action_item_params = {
+            title:  single_assignment_params.fetch(:title),
+            description: single_assignment_params.fetch(:description),
+            due_date: due_date,
+            category: single_assignment_params.fetch(:category),
+            file: single_assignment_params.fetch(:file),
+        }
+        if single_assignment_params.fetch(:file).eql?("null") || single_assignment_params.fetch(:file).eql?("undefined") || !single_assignment_params.fetch(:file).present?
```

Comment:
this looks pretty messy as an if statement. I'd prefer this just be an appropriate named private helper function. I also don't know if all the eqls are necessary. In what instances does it return as "null" vs "undefined" vs not being present?

Reply:
I think there will always be something there it is either a file, "null", or "undefined". "null" comes from not selecting a file. "undefined" comes from opening the file opener and then closing it.